### PR TITLE
GCS on Local Closes #1161

### LINKED ISSUE #1161: Add back GCS support to local backend
The local backend used to have GCS file support. Add this back in in order to hit feature parity. 

At the current moment it is believed that this is not a huge task. If it turns out to be a giant morass please speak up and let's evaluate

### PR DESCRIPTION
**Pull request by Horneth**
Associated centaur test: https://github.com/broadinstitute/centaur/pull/89

### CODE REVIEW

File: supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalImplicits.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import cromwell.core.WorkflowOptions
+import cromwell.filesystems.gcs.GoogleAuthMode
+import cromwell.filesystems.gcs.GoogleAuthMode.GoogleAuthOptions
+
+import scala.util.Try
+
+object LocalImplicits {
+
+  // TODO how to harmonize this ? Already exists in JES but needs gcs dependency - so can't be in core or backend at the moment
```
Comment (mcovarr) [inline supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalImplicits.scala]: yeah I struggled with this problem before resorting to copy/paste.  😄

**Comment by scottfrazer** [comment]
👍

File: supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalInitializationActor.scala [hunk 2]
```diff
@@ -1,43 +1,59 @@
...
     DockerKey -> wdlTypePredicate(valueRequired = false, WdlStringType.isCoerceableFrom),
     FailOnStderrKey -> wdlTypePredicate(valueRequired = false, WdlBooleanType.isCoerceableFrom),
     ContinueOnReturnCodeKey -> continueOnReturnCodePredicate(valueRequired = false)
   )
 
-  private val workflowPaths = new WorkflowPaths(workflowDescriptor, configurationDescriptor.backendConfig, None)
+  private val fileSystems = {
+    val maybeGcs = {
+      localConfiguration.gcsAuthMode map { authMode =>
+        val storage =authMode.buildStorage(workflowDescriptor.workflowOptions.toGoogleAuthOptions, localConfiguration.googleConfig)
```
Comment (geoffjentry) [inline supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalInitializationActor.scala]: some spacing issues w/ the `=`

Also I'd love to not have 8 line val assignments but feel free to ignore me on that one

**Comment by kshakir** [comment]
LGTM 👍

**Comment by mcovarr** [comment]
Looks good.  When the SGE backend makes its much anticipated return there are some bits here that could probably be refactored to be shared.  :+1:

**Pull request merged**